### DISCUSSION TITLE
Using axum_login on nested routes.

### DISCUSSION THREAD

**Discussion by aarondyejr** [maxcountryman/axum-login#237] [Q&A] [answered]
Hello all, I am having some issues myself with making "protected" routes that won't interfere with non-protected routes. This is what I tried first:
```rs
        let protected_routes = Router::new()
            .nest("/users", routes::user::router().await)
            .route_layer(login_required!(Backend, login_url = "/"));

        let auth_routes = Router::new().nest("/auth", routes::auth::router().await);

        let api_routes = Router::new().merge(auth_routes).merge(protected_routes);
```
Problem is I can login, I can logout, but when I visit `/users/@me` (@me is a route on /users) I get a 404, as if the route isn't even being reached. I confirmed this by dumping "Hello, World" at that route and its still 404. I am not sure if axum_login is interfering at this point, or if my setup is wrong. Either way hopefully someone can give me the answer.

**Comment by aarondyejr** [comment]
I am also noticing that whenever I "logout" I am getting `2024-07-03T03:49:50.280683Z  WARN call:call:get_record: tower_sessions_core::session: possibly suspicious activity: record not found in store` in my console. As well as sometimes `2024-07-03T03:48:57.688271Z  WARN call:call:delete: tower_sessions_core::session: called delete with no session id`

**Reply by aarondyejr** [reply, marked as answer]
Right, so I fixed the issue, it was due to a typo I had in my surrealdb select query to grab the user. Everything is working fine now, but I do still have a question remaining. It seems when I logout `2024-07-03T04:07:37.485521Z  WARN call:call:get_record: tower_sessions_core::session: possibly suspicious activity: record not found in store` this gets posted, and yet it deletes the session, and the cookie

**Reply by maxcountryman** [reply]
Are you seeing that message after restarting the server? That will appear when the browser sends a request with a cookie that's no longer valid (which could be the case after restarting your server, depending on how your app is storing sessions).

**Reply by aarondyejr** [reply]
It does happen when I restart the server, but also it happens even during the server running. It sets the session, I hit the logout endpoint and that message appears despite still deleting the session, cookie and all.

**Reply by maxcountryman** [reply]
That message indicates the user agent has provided a deleted session cookie on the request; you can see the call site in `tower-sessions`.

Without seeing your code, it's going to be difficult to help.

Can you reproduce what you're seeing using one of the examples in this repo?

**Reply by aarondyejr** [reply]
I seem to have fixed it, but I do have a big concern on whether or not this will impede if the backend goes down or not, those sessions become invalid. Which at scale is not a good thing having x amount of users having to log back in. It may very well be the store I am using though so I might have to take it up with them.

**Reply by maxcountryman** [reply]
It really depends on what you're using as a backend.

Generally speaking, sessions should be persisted between server restarts (so are stored in a durable way) and so starting and stopping the server itself would not invalidate a session.

**Reply by aarondyejr** [reply]
I see, then I guess I will have to look more into it, as it seems like it is invalidating the sessions (as when I visit a protected route it throws a 401 despite me having a valid session & cookie)

**Reply by maxcountryman** [reply]
What session store are you using?

**Reply by aarondyejr** [reply]
surrealdb

**Reply by maxcountryman** [reply]
Unfortunately I don't know much about Surreal.

**Reply by aarondyejr** [reply]
Fair enough! I will keep monitoring my situation and see if its not some fluke and if its an actual bug I will open an issue with the maintainer of the surrealdb store library :) Thanks a lot for your help max!
